decayAmplitude/waveDescription.cc

```diff
@@ -868,23 +1004,117 @@ waveDescription::mapMassDependenceType(const Setting* massDepKey)
 		}
 		const int length = bounds->getLength();
 		if (length != 2) {
-			printErr << "bounds do not have the required length (expected: 2, found: " << length << ")." << std::endl;
+			printErr << "bounds do not have the required length (expected: 2, found: " << length << ")." << endl;
 			throw;
 		}
 		const double mMin = (*bounds)[0];
 		const double mMax = (*bounds)[1];
 		if (mMin > mMax) {
-			printErr << "bounds are not ordered: mMin(" << mMin << ") > mMax(" << mMax << ")." << std::endl;
+			printErr << "bounds are not ordered: mMin(" << mMin << ") > mMax(" << mMax << ")." << endl;
 			throw;
 		}
-		massDep = createbinnedMassDependence(mMin, mMax);
+		massDep = createBinnedMassDependence(mMin, mMax);
+	}
+	else if (massDepType == "sawtooth") {
+		const Setting* bounds = findLibConfigList(*massDepKey, "bounds" , false);
+		if (not bounds) {
+			printErr << "no bounds given for sawtooth mass dependence." << endl;
+			throw;
+		}
+		const int length = bounds->getLength();
+		if (length != 2) {
+			printErr << "bounds do not have the required length (expected: 2, found: " << length << ")." << endl;
+			throw;
+		}
+		const double mMin = (*bounds)[0];
+		const double mMax = (*bounds)[1];
+		if (mMin > mMax) {
+			printErr << "bounds are not ordered: mMin(" << mMin << ") > mMax(" << mMax << ")." << endl;
+			throw;
+		}
+		massDep = createSawtoothMassDependence(mMin, mMax);
+	}
+	else if (massDepType == "polynomial") {
+		double mMin = 0.;
+		bool mMinFound = massDepKey->lookupValue("mMin", mMin);
+		double mMax = 0.;
+		bool mMaxFound = massDepKey->lookupValue("mMax", mMax);
+		if (mMinFound != mMaxFound) {
+			printErr << "something wrong with the setting of the mass boundaries. Aborting..." << endl;
+			throw;
+		}
+		if (not mMinFound) {
+			mMin = -1.; // No mass boundaries were given. So no scaling is set.
+			mMax =  1.; // This is achieved by setting mMin = -1. and mMax = 1.
+		}
+		const Setting* realCoefficients = findLibConfigList(*massDepKey, "realCoefficients" , false);
+		if (not realCoefficients) {
+			printErr << "no real coefficients given for polynomial mass dependence" << endl;
+			throw;
+		}
+		const int length = realCoefficients->getLength();
+		bool hasImagCoefficients = true;
+		const Setting* imagCoefficients = findLibConfigList(*massDepKey, "imagCoefficients" , false);
+		if (not imagCoefficients) {
+			hasImagCoefficients = false;
+		} else {
+			if (length != imagCoefficients->getLength()) {
+				printErr << "different number of real and imag coefficients given." << endl;
+				throw;
+			}
+		}
+		vector<complex<double> > coefficients;
+		for (int i = 0; i < length; ++i) {
+			if (hasImagCoefficients) {
+				coefficients.push_back(complex<double>((*realCoefficients)[i], (*imagCoefficients)[i]));
+			} else {
+				coefficients.push_back(complex<double>((*realCoefficients)[i], 0.));
+			}
+		}
+		massDep = createPolynomialMassDependencePtr(coefficients, mMin, mMax);
+	}
+	else if (massDepType == "complexExponential") {
```

Comment:
can't one use the bounds field of binned and sawtooth here as well (instead of mMin and mMax)?